src/Raven.Server/Web/Studio/LicenseHandler.cs

```diff
@@ -106,12 +125,29 @@ public async Task RenewLicense()
                 return;
             }
 
-            using (var context = JsonOperationContext.ShortTermSingleUse())
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
             using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
             {
                 var renewLicense = await ServerStore.LicenseManager.RenewLicense();
                 context.Write(writer, renewLicense.ToJson());
             }
         }
     }
+    
+    public class LicenseConfigurationSettings : IDynamicJson
+    {
+        public bool RenewSettings { get; set; }
```

Comment:
names in LicenseConfigurationSettings are misleading,

Reply:
same answer as above

Reply:
I concur with Marcin, I think it would be better to match the configuration properties e.g. CanActivate, CanRenew (rename the property and add a 2nd key), etc

Reply:
like I wrote: when I see property with name: SomethingCount, I expect number. When I see SomethingList I expect array/list. But when I see SomethingSettings then boolean type is unexpected.

Reply:
Changed the endpoint to use the original `LicenseConfiguration ` class